include/rmm/cuda_device.hpp

```diff
@@ -102,6 +103,52 @@ inline int get_num_cuda_devices()
   return num_dev;
 }
 
+/**
+ * @brief Returns the available and total device memory in bytes for the current device
+ *
+ * @return The available and total device memory in bytes for the current device as a std::pair.
+ */
+inline std::pair<std::size_t, std::size_t> available_device_memory()
```

Comment:
**suggestion/nit** At point of use of this function, the user must "just remember" that the first entry in the pair is the free and the second is the total device memory. How about instead:

```
struct mem_info {
    std::size_t free;
    std::size_t total;
}

inline mem_info available_device_memory()
{
   ...
}
```
Instead?

Reply:
Often one ignores one of the return values e.g. with 

```
auto [[maybe_unused]] [free, _] = available_device_memory();
```

Does the struct approach allow this?

Aside: I always hated that cudaMemGetInfo returns two things. There should be two separate functions.

Reply:
Looks like it does, yes https://gcc.godbolt.org/z/eqzaG8cY9

Reply:
I don't think this is a blocking suggestion, so I'll let you choose whether or not to make this change, approving as is.

Reply:
Thanks.